### PR TITLE
fix(state-sync): Config option to provide a GCS credentials file in the config file

### PR DESCRIPTION
**Pull request by nikurt**
Also adjust verbosity of `state-parts` logging

### CODE REVIEW

File: nearcore/src/config.rs [hunk 1]
```diff
@@ -350,8 +350,10 @@ pub struct Config {
     /// chunks and underutilizing the capacity of the network.
     #[serde(default = "default_transaction_pool_size_limit")]
     pub transaction_pool_size_limit: Option<u64>,
-    /// If a node needs to upload state parts to S3
+    /// If a node needs to upload state parts to S3.
     pub s3_credentials_file: Option<String>,
+    /// If a node needs to upload state parts to GCS.
+    pub gcs_credentials_file: Option<String>,
```
Comment (wacban) [inline nearcore/src/config.rs:356]: nit1: It feels like those should be exclusive. How about putting them in an enum or a runtime check?
nit2: You can also check if the credentials for state_sync.dump.location are Some. 
nit3: How about moving those to the state sync config? Please disregard if used elsewhere.

**Review by wacban — APPROVED** [review]
lgtm

File: nearcore/src/state_sync.rs [hunk 2]
```diff
@@ -42,17 +43,30 @@ pub fn spawn_state_sync_dump(
         return Ok(None);
     };
     tracing::info!(target: "state_sync_dump", "Spawning the state sync dump loop");
+    println!("Spawning the state sync dump loop");
```
[the remark this answers is not in the document]
Comment (nikurt) [reply nearcore/src/state_sync.rs:46]: Done

File: nearcore/src/config.rs (hunk 1 above)
Comment (nikurt) [reply nearcore/src/config.rs:356]: nit1: Not really. I can imagine a node that has permissions to write to both locations. Anyway, moved the field to DumpConfig, and now it's exclusive.
nit2: Not sure what you meant.
nit3: Moved it to state_sync.dump.credentials_file.

Comment (wacban) [reply nearcore/src/config.rs]: nit1: ok

nit2: 
```
if location == gcp && gcs_credentials_file.is_none() { return error("missing credentials"); }
// not a biggie
```

nit3: thanks

**Pull request merged**